loadbalancer/zk/client.go

```diff
@@ -0,0 +1,157 @@
+package zk
+
+import (
+	"errors"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/eapache/channels"
+	"github.com/samuel/go-zookeeper/zk"
+
+	"github.com/go-kit/kit/log"
+)
+
+// DefaultACL is the default ACL to use for creating znodes
+var DefaultACL = zk.WorldACL(zk.PermAll)
+
+const (
+	// ConnectTimeout is the default timeout to establish a connection to a
+	// ZooKeeper node.
+	ConnectTimeout = 2 * time.Second
+	// SessionTimeout is the default timeout to keep the current ZooKeeper
+	// session alive during a temporary disconnect.
+	SessionTimeout = 5 * time.Second
+)
+
+// Client is a wrapper around a lower level ZooKeeper client implementation.
+type Client interface {
+	// GetEntries should query the provided path in ZooKeeper, place a watch on
+	// it and retrieve data from its current child nodes.
+	GetEntries(path string) ([]string, channels.SimpleOutChannel, error)
```

Comment:
As we discussed on Slack, I think the interface is still valuable for testing. See the [consul.Client interface](https://github.com/go-kit/kit/blob/d81f9d953781636e3fd67c9fc4eab1210f3437b3/loadbalancer/consul/client.go#L6-L8) which is [accepted by the Publisher](https://github.com/go-kit/kit/blob/d81f9d953781636e3fd67c9fc4eab1210f3437b3/loadbalancer/consul/publisher.go#L32) and [faked for the tests](https://github.com/go-kit/kit/blob/d81f9d953781636e3fd67c9fc4eab1210f3437b3/loadbalancer/consul/publisher_test.go#L164).